plugin/src/main/resources/kotlin/data_class.mustache

```diff
@@ -4,6 +4,7 @@
  * @property {{{name}}} {{description}}
 {{/vars}}
  */
+@JsonClass(generateAdapter = true)
```

Comment:
You probably need to add the same annotation on top of the `enum class` below

Reply:
For some reason I did not notice issues if I would not do it, I'll try to debug it a bit longer to see that the assumptions are verified and if we need to do it then we definitely need to create a test-scenario for it in order to make sure that we don't introduce a regression.

Side note: it's highly probable that I've not seen an issue as moshi-kotlin (and so reflection was still present)

Reply:
I've tried adding `@JsonClass(generateAdapter = true)` to the enum class and annotation processing fails with the following error `code gen for enums is not supported or necessary`.

Defining it as `@JsonClass(generateAdapter = false)` (as in #68) is doable but essentially no-op, so I would rather prefer not generating it at all.

Feel free to re-open it if you think that we need to annotate it anyway.